WooCommerce/Classes/ViewRelated/Products/Edit Product/BottomSheetListSelector/BottomSheetProductType.swift

```diff
@@ -199,6 +198,19 @@ public enum BottomSheetProductType: Hashable, Identifiable {
         }
     }
 
+    /// Product creation category (for UI purposes).
+    ///
+    var productCreationCategory: ProductCreationCategory {
+        switch self {
+        case .simple, .blank, .variable:
+            return .standard
+        case .subscription, .variableSubscription:
+            return .subscription
+        case .custom, .grouped, .affiliate:
+            return .other
+        }
+    }
```

Comment:
Suggestion: you can avoid having to filter and match the category types if you replace this with a helper variable in `ProductCreationCategory` instead:

```Swift
extension ProductCreationCategory {
    var productTypes: BottomSheetProductType {
        // return the respective types for each category.
    }
}
```

Then in `ManualProductTypeOptions` you can get rid of `groupedProductTypes` and build the view by directly traversing the categories:

```Swift
ForEach(BottomSheetProductType.ProductCreationCategory.allCases, id: \.self) { category in
     ForEach(category.productTypes) { type in
          // TODO
     }
}
```

Reply:
I did some refactor on https://github.com/woocommerce/woocommerce-ios/pull/12273/commits/f1b37ed71271104480c29d1be2724a266c109d5b based on your ideas here and on the suggestion above, but it doesn't fully implement your suggestion as I understood it.

The main thing is that the `ManualProductTypeOptions` receives `supportedProductTypes` that contains only the supported product types (calculated in  `presentActionSheetWithAI()` in `AddProductCoordinator`), so filtering things based on it is inevitable.